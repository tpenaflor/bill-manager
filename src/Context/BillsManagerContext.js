import React from 'react'

const BillsManagerContext = React.createContext()

export default BillsManagerContext