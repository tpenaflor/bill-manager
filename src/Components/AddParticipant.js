
import React, {useState, useContext} from 'react'
import BillsManagerContext from '../Context/BillsManagerContext'
import { v4 } from 'uuid';

const AddParticipant = () => {
    const {partDispatch} = useContext(BillsManagerContext)
    const [name, setName] = useState('')

    const addParticipant = (e) => {
        e.preventDefault()
        partDispatch({type:"ADD_PARTICIPANT", participant:{id:v4(), name}})
        setName('')
      }

  return (
    <form onSubmit={addParticipant}>
      <input value ={name} 
        onChange={(e)=> setName(e.target.value.trim())}
        placeholder='Name'  />
      <button>Add Participant</button>
    </form>
  )
}

export default AddParticipant