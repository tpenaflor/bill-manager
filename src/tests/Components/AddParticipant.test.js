import React from 'react'
import {shallow} from 'enzyme'
import { shallowToJson  } from 'enzyme-to-json';

import AddParticipant from '../../Components/AddParticipant'
import * as BillsManagerContext from '../../Context/BillsManagerContext'

let partDispatch, wrapper

beforeEach( ()=> {
    partDispatch = jest.fn()
    const contextValues = {
      partDispatch
    }

    jest.spyOn(BillsManagerContext, 'useBillsManagerContext')
        .mockImplementation(() => contextValues)

    wrapper = shallow(<AddParticipant />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
})

afterEach(()=>{
    expect(shallowToJson(wrapper)).toMatchSnapshot()
})

test('render add participants', ()=> {
    const name = "participant name"
    wrapper.find('input').at(0).simulate('change', {target:{value: name}})
    expect(shallowToJson(wrapper)).toMatchSnapshot()
    
    wrapper.find('form').simulate('submit', {preventDefault:()=>{}})

    expect(partDispatch).toHaveBeenLastCalledWith({
        type:"ADD_PARTICIPANT", 
        participant: {
          id : expect.any(String),
          name
        }})

})
