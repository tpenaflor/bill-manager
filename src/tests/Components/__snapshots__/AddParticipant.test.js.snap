// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render add participants 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Name"
    value=""
  />
  <button>
    Add Participant
  </button>
</form>
`;

exports[`render add participants 2`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Name"
    value="participant name"
  />
  <button>
    Add Participant
  </button>
</form>
`;

exports[`render add participants 3`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Name"
    value=""
  />
  <button>
    Add Participant
  </button>
</form>
`;
